File: _Exercises/SampleMySQLLiteApp_Solution/app/src/main/java/com/mobdeve/tighee/samplesqliteapp/MyViewHolder.java
package com.mobdeve.tighee.samplesqliteapp;

import android.net.Uri;
import android.util.Log;
import android.view.View;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.squareup.picasso.Picasso;

public class MyViewHolder extends RecyclerView.ViewHolder {
    private TextView nameTv, numberTv;
    private ImageView imageIv;
    private ImageButton deleteBtn, editBtn;

    public MyViewHolder(View itemView) {
        super(itemView);

        this.nameTv = itemView.findViewById(R.id.nameTv);
        this.numberTv = itemView.findViewById(R.id.numberTv);
        this.imageIv = itemView.findViewById(R.id.imageIv);
        this.deleteBtn = itemView.findViewById(R.id.deleteBtn);
        this.editBtn = itemView.findViewById(R.id.editBtn);
    }

    public void bindData(Contact c) {
        this.nameTv.setText(c.getLastName() + ", " + c.getFirstName());
        this.numberTv.setText(c.getNumber());
        Log.d("TAG", "bindData: " + c.toString());
        Picasso.get()
            .load(c.getImageUri())
            .into(this.imageIv);
    }

    public void setDeleteBtnOnClickListener(View.OnClickListener onClickListener) {
        this.deleteBtn.setOnClickListener(onClickListener);
    }

    public void setEditBtnOnClickListener(View.OnClickListener onClickListener) {
        this.editBtn.setOnClickListener(onClickListener);
    }
}

File: _Exercises/SampleMySQLLiteApp/app/src/main/java/com/mobdeve/tighee/samplemysqlliteapp/MyViewHolder.kt
package com.mobdeve.tighee.samplemysqlliteapp

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso

class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private val nameTv: TextView
    private val numberTv: TextView
    private val imageIv: ImageView

    init {
        nameTv = itemView.findViewById(R.id.nameTv)
        numberTv = itemView.findViewById(R.id.numberTv)
        imageIv = itemView.findViewById(R.id.imageIv)
    }

    fun bindData(c: Contact) {
        nameTv.text = c.lastName + ", " + c.firstName
        numberTv.text = c.number
        Picasso.get()
            .load(c.imageUri)
            .into(imageIv)
    }
}